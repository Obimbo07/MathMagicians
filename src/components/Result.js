import React from 'react';
import PropTypes from 'prop-types';

const Results = ({ result }) => <div>{ result }</div>;

Results.defaultProps = {
  result: 0,
};

Results.propTypes = {
  result: PropTypes.string,
};

export default Results;
