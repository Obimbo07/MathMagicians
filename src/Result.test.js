import React from 'react';
import { render } from '@testing-library/react';
import Result from './components/Result';

describe('Results component', () => {
  it('renders correctly with a number value', () => {
    const { asFragment } = render(<Result value={42} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with a different number value', () => {
    const { asFragment } = render(<Result value={123} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
