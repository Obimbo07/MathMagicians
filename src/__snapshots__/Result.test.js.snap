// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component renders correctly with a different number value 1`] = `
<DocumentFragment>
  <div>
    0
  </div>
</DocumentFragment>
`;

exports[`Results component renders correctly with a number value 1`] = `
<DocumentFragment>
  <div>
    0
  </div>
</DocumentFragment>
`;
